Modules/Core/Common/include/itkPathType.h

```diff
@@ -0,0 +1,29 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         https://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#ifndef itkPathType_h
+#define itkPathType_h
+
+#include <filesystem>
+
+namespace itk
+{
+using PathType = std::filesystem::path;
```

Comment:
What is the need for this indirection?

Reply:
I am adding a similar type in SimpleITK to transition from std::string to std::filesystem::path which is being conditioned to the preprocessor (maybe).  It also made wrapping with SWIG a little easier. 

It is shorter?

Reply:
Personally I would prefer to just use `std::filesystem::path`. (Unless it would need to be configurable, as in SimpleITK.) ITK already has so many typedefs (or type aliases, if you like). When doing code analysis, I often lose time, searching for the definition of a typedef. It doesn't always make the code more readable to have all those typedefs.